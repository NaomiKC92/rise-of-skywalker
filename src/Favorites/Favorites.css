.bb8-image {
  width: 200px;
}
