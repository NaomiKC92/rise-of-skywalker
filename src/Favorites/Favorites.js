import React from 'react';
import './Favorites.css';
import isFavoriteFalse from '../../src/images/isFavoriteFalse.svg';
import CharacterCard from '../CharacterCard/CharacterCard';

const Favorites = ({
  favoriteCharacters,
  setupFavoriteCharacters,
  favoriteCharactersMap
}) => {
  if (favoriteCharacters.length === 0) {
    return (
      <React.Fragment>
        <h2>
          You can favorite any of the characters you like by pressing the cute
          BB8 icon or the favorite button on each character!
        </h2>
        <img className='bb8-image' src={isFavoriteFalse} alt='BB8 Icon' />
      </React.Fragment>
    );
  }

  const allFavoriteCharacters = favoriteCharacters.map((character, i) => {
    return (
      <CharacterCard
        character={character}
        setupFavoriteCharacters={setupFavoriteCharacters}
        favoriteCharactersMap={favoriteCharactersMap}
        key={i}
      />
    );
  });

  return (
    <div>
      <div>{allFavoriteCharacters}</div>
    </div>
  );
};

export default Favorites;
