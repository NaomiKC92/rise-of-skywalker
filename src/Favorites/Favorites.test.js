import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './Favorites';

describe('FAVORITES', () => {
  let wrapper;

  const mockFavoriteCharacters = [];
  const mockSetupFavoriteCharacters = jest.fn();
  const mockFavoriteCharactersMap = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Favorites
        favoriteCharacters={mockFavoriteCharacters}
        setupFavoriteCharacters={mockSetupFavoriteCharacters}
        favoriteCharactersMap={mockFavoriteCharactersMap}
      />
    );
  });

  it('should match the Favorites Component Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
