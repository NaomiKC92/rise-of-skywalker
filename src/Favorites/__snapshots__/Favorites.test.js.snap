// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAVORITES should match the Favorites Component Snapshot 1`] = `
<Fragment>
  <h2>
    You can favorite any of the characters you like by pressing the cute BB8 icon or the favorite button on each character!
  </h2>
  <img
    alt="BB8 Icon"
    className="bb8-image"
    src="isFavoriteFalse.svg"
  />
</Fragment>
`;
